src/roi/Roi.ts

```diff
@@ -392,6 +394,17 @@ export class Roi {
     });
   }
 
+  get ellipse(): Ellipse {
+    return this.#getComputed('ellipse', () => {
+      let ellipse = getEllipse(this, 1);
+      if (ellipse.surface !== this.surface) {
+        const scaleFactor = Math.sqrt(this.surface / ellipse.surface);
+        ellipse = getEllipse(this, scaleFactor);
```

Comment:
I would have never done it like that because this process (calculate the ellipse) is CPU intensive. Please use as second parameter the surface `getEllipse(this, surface)` and `getEllipse` can calculate the scaleFactor internally.